lib/lightning/accounts.ex

```diff
@@ -24,33 +45,54 @@ defmodule Lightning.Accounts do
       # coveralls-ignore-stop
     end)
 
-    # Remove credentials
-    Credentials.list_credentials_for_user(id)
-    |> Enum.each(&Credentials.delete_credential/1)
-
-    # Revoke access to projects
+    # Delete all project_users
     Projects.get_projects_for_user(%User{id: id})
     |> Repo.preload(:project_users)
-    |> Enum.each(fn p ->
-      Projects.update_project(
-        p,
-        %{
-          "project_users" => %{
-            "0" => %{
-              "delete" => "true",
-              "user_id" => id,
-              "id" =>
-                Enum.find(p.project_users, fn pu -> pu.user_id == id end)
-                |> Map.get(:id)
-            }
-          }
-        }
-      )
+    |> Enum.map(fn p ->
+      {p, %{"project_users" => project_users_to_params(p.project_users, id)}}
     end)
+    |> Enum.each(fn {project, param} ->
```

Comment:
Why are we using params to work with changesets here? Usually params are used to work with requests from outside the application, like forms and http etc.

This should do the same thing.

```elixir
Ecto.assoc(%User{id: id}, :project_users) |> Repo.delete_all()
```

There's nothing wrong with working with params, they can be very useful when generating new records from nothing. But in this case, it can be solved with a query.

Reply:
Done